--- a/src/sort.rs
+++ b/src/sort.rs
@@ -2591,10 +2591,22 @@
 }
 
 fn sort_by<'a>(unsorted: &mut Vec<Line<'a>>, settings: &GlobalSettings, line_data: &LineData<'a>) {
-    if settings.stable || settings.unique {
-        unsorted.par_sort_by(|a, b| compare_by(a, b, settings, line_data, line_data));
-    } else {
-        unsorted.par_sort_unstable_by(|a, b| compare_by(a, b, settings, line_data, line_data));
+    #[cfg(target_os = "wasi")]
+    {
+        if settings.stable || settings.unique {
+            unsorted.sort_by(|a, b| compare_by(a, b, settings, line_data, line_data));
+        } else {
+            unsorted.sort_unstable_by(|a, b| compare_by(a, b, settings, line_data, line_data));
+        }
+    }
+
+    #[cfg(not(target_os = "wasi"))]
+    {
+        if settings.stable || settings.unique {
+            unsorted.par_sort_by(|a, b| compare_by(a, b, settings, line_data, line_data));
+        } else {
+            unsorted.par_sort_unstable_by(|a, b| compare_by(a, b, settings, line_data, line_data));
+        }
     }
 }
 
