--- a/src/uname.rs
+++ b/src/uname.rs
@@ -76,12 +76,31 @@ impl UNameOutput {
 
         let nodename = (opts.nodename || opts.all).then(|| uname.nodename().to_owned());
 
+        #[cfg(target_os = "wasi")]
+        let kernel_name = (opts.kernel_name || opts.all || none)
+            .then(|| OsString::from("Linux"));
+
+        #[cfg(target_os = "wasi")]
+        let nodename = (opts.nodename || opts.all)
+            .then(|| OsString::from("agentos"));
+
         let kernel_release = (opts.kernel_release || opts.all).then(|| uname.release().to_owned());
 
         let kernel_version = (opts.kernel_version || opts.all).then(|| uname.version().to_owned());
 
         let machine = (opts.machine || opts.all).then(|| uname.machine().to_owned());
 
         let os = (opts.os || opts.all).then(|| uname.osname().to_owned());
+
+        #[cfg(target_os = "wasi")]
+        let kernel_release = (opts.kernel_release || opts.all)
+            .then(|| OsString::from("0.0.1-agentos"));
+        #[cfg(target_os = "wasi")]
+        let kernel_version = (opts.kernel_version || opts.all)
+            .then(|| OsString::from("#1 AgentOS Linux-in-WASM"));
+        #[cfg(target_os = "wasi")]
+        let machine = (opts.machine || opts.all).then(|| OsString::from("wasm32"));
+        #[cfg(target_os = "wasi")]
+        let os = (opts.os || opts.all).then(|| OsString::from("GNU/Linux"));
 
         // This option is unsupported on modern Linux systems
